src/components/AnswersProvider/AnswersProvider.test.js

```diff
@@ -51,4 +80,135 @@ describe('QuestionsProvider', () => {
 
     expect(mockedOnClick).toHaveBeenCalledWith(expectedState);
   });
+
+  it('synchronize the list of answers in batch', async () => {
+    render(
+      <AnswersProvider {...props}>
+        <AnswersContext.Consumer>
+          {({ answers, updateAnswers }) => (
+            <>
+              <button
+                onClick={() =>
+                  updateAnswers({
+                    [Object.keys(answers).length]: { answer: 'DT' },
+                  })
+                }
+              >
+                Add answer
+              </button>
+            </>
+          )}
+        </AnswersContext.Consumer>
+      </AnswersProvider>,
+    );
+
+    const addAnswerButton = await screen.findByRole('button', {
+      name: 'Add answer',
+    });
+
+    fireEvent.click(addAnswerButton);
+    expect(questionsService.syncAnswers).not.toBeCalled();
+
+    fireEvent.click(addAnswerButton);
+    expect(questionsService.syncAnswers).not.toBeCalled();
+
+    fireEvent.click(addAnswerButton);
+    expect(questionsService.syncAnswers).not.toBeCalled();
+
+    fireEvent.click(addAnswerButton);
+    expect(questionsService.syncAnswers).not.toBeCalled();
+
+    fireEvent.click(addAnswerButton);
+    expect(questionsService.syncAnswers).toBeCalledWith({
+      firebase: props.firebase,
+      user: props.user,
+      currentUser: props.currentUser,
+      answers: {
+        ...storedAnswers,
+        2: { answer: 'DT' },
+        3: { answer: 'DT' },
+        4: { answer: 'DT' },
+        5: { answer: 'DT' },
+        6: { answer: 'DT' },
+      },
+    });
+
+    questionsService.syncAnswers.mockClear();
+
+    fireEvent.click(addAnswerButton);
+    expect(questionsService.syncAnswers).not.toBeCalled();
+
+    fireEvent.click(addAnswerButton);
+    expect(questionsService.syncAnswers).not.toBeCalled();
+
+    fireEvent.click(addAnswerButton);
+    expect(questionsService.syncAnswers).not.toBeCalled();
+
+    fireEvent.click(addAnswerButton);
+    expect(questionsService.syncAnswers).not.toBeCalled();
+
+    fireEvent.click(addAnswerButton);
+    expect(questionsService.syncAnswers).toBeCalledWith({
+      firebase: props.firebase,
+      user: props.user,
+      currentUser: props.currentUser,
+      answers: {
+        ...storedAnswers,
+        2: { answer: 'DT' },
+        3: { answer: 'DT' },
+        4: { answer: 'DT' },
+        5: { answer: 'DT' },
+        6: { answer: 'DT' },
+        7: { answer: 'DT' },
+        8: { answer: 'DT' },
+        9: { answer: 'DT' },
+        10: { answer: 'DT' },
+        11: { answer: 'DT' },
+      },
+    });
```

Comment:
Você poderia iterar em um for vazio e disparar tudo e depois fazer a asserção +- assim:

```js
    expect(questionsService.syncAnswers).toBeCalledTimes(2);
``
`

Reply:
Nice approach!